### PR TITLE
HBASE-23710 - Priority configuration for system coprocessors

### PR DESCRIPTION
**Pull request by gjacoby126**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 32s |  master passed  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  8s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 41s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 40s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 39s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m  7s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 40s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 38s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 152m 51s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 210m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1077 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux c80b8f1d84fa 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1077/out/precommit/personality/provided.sh |
| git revision | master / 569ac1232a |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/1/testReport/ |
| Max. process+thread count | 4411 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/CoprocessorHost.java [hunk 1]
```diff
@@ -143,8 +143,16 @@ protected void loadSystemCoprocessors(Configuration conf, String confKey) {
     if (defaultCPClasses == null || defaultCPClasses.length == 0)
       return;
 
-    int priority = Coprocessor.PRIORITY_SYSTEM;
+    int currentSystemPriority = Coprocessor.PRIORITY_SYSTEM;
     for (String className : defaultCPClasses) {
+      String[] classNameAndPriority = className.split("\\|");
```
Comment (virajjasani) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/CoprocessorHost.java]: May be good to update doc for `hbase.coprocessor.regionserver.classes` in section `Restricting Coprocessor Usage` of cp.adoc?

Comment (gjacoby126) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/CoprocessorHost.java]: Sure, I added some notes to cp.adoc and updated the PR, @virajjasani .

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 13s |  master passed  |
| +1 :green_heart: |  compile  |   3m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   2m 13s |  master passed  |
| +0 :ok: |  refguide  |   5m 33s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  shadedjars  |   4m 35s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 23s |  master passed  |
| +0 :ok: |  spotbugs  |   4m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |  18m 23s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  2s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 11s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +0 :ok: |  refguide  |   5m 41s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 35s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  findbugs  |  18m 46s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 206m 43s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 24s |  The patch does not generate ASF License warnings.  |
|  |   | 317m  2s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.snapshot.TestExportSnapshotNoCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1077 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile refguide |
| uname | Linux c3dd03ff9615 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1077/out/precommit/personality/provided.sh |
| git revision | master / 00e64d83e8 |
| Default Java | 1.8.0_181 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/2/artifact/out/branch-site/book.html |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/2/artifact/out/patch-site/book.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/2/testReport/ |
| Max. process+thread count | 5537 (vs. ulimit of 10000) |
| modules | C: hbase-server . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1077/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by gjacoby126** [comment]
The test failure doesn't appear to be related to this change (they're all MapReduce export tests), and no code in this patch has changed since the last passing run. The only subsequent change was to docs.

**Comment by gjacoby126** [comment]
Thanks for the review, @virajjasani . If there's no other changes you'd like, could you please commit when you get the chance? I've uploaded a backport to branch-1 to make that easier.

**Comment by virajjasani** [comment]
+1, sure let me commit this today. Test failure is irrelevant anyways.

**Pull request merged**